src/lib/components/Websocket.svelte

```diff
@@ -1,33 +1,28 @@
 <script lang="ts">
 	import { socket } from '$lib/stores/stateStore';
-	import { svgs, type Svg } from '$lib/stores/svgStore';
+	import { svgs } from '$lib/stores/svgStore';
 	import { onMount } from 'svelte';
+	import type { Edit, Remove, Svg } from '$lib/types';
 
 	onMount(() => {
 		$socket.on('edit', handleEdit);
 		$socket.on('remove', handleRemove);
 	});
 
 	// TODO: Mads, optionals i interface.
-	function handleEdit(data: any) {
+
+	function handleEdit(data: Edit) {
 		// check if the command already exists
 		const commandIndex = $svgs.findIndex((svg: Svg) => svg.commandId === data.commandId);
 
 		// if it exists, modify it
-		// HACK: Maybe rewrite to loop through the keys of data instead
-		if (commandIndex !== -1) {
-			// only modify the property if it exists on the data object
-			if ('x' in data) {
-				$svgs[commandIndex].x = data.x;
-			}
-			if ('y' in data) {
-				$svgs[commandIndex].y = data.y;
-			}
-			if ('svg' in data) {
-				$svgs[commandIndex].svg = data.svg;
-			}
+		if (commandIndex === -1) {
+			$svgs[commandIndex].placement = data.placement ?? $svgs[commandIndex].placement;
+			$svgs[commandIndex].svgString = data.svgString ?? $svgs[commandIndex].svgString;
 			return;
 		}
+
+		if (!data.svgString || !data.placement) return;
```

Comment:
Er det en gyldig tilstand programmet kan have, hvor enten svgString eller placement ikke er der?

Hvis ikke, så skjuler vi bare en fejlagtig tilstand. Der burde vel være en exception eller allermindst en log, hvis det er en fejlagtig tilstand

Reply:
Dette er fordi at det sendes igennem socketio, som ikke har indbygget input sanitering. Derfor skal det altid tjekkes, for at en bruger ikke kan forbinde til et socket f.eks. i postman og sende en event der crasher vores server